### PR TITLE
Yield out potentional slot instructions when rendering dynamic tags

### PR DESCRIPTION
**Pull request by matthewp**
## Changes

- Dynamic tags take a different code-path inside of renderComponent
 - If there are slot instructions (basically this is the scripts to render), make sure to yield those out before the HTML.
 - Fixes https://github.com/withastro/astro/issues/4971

## Testing

- Fixture test added

## Docs

N/A, bug fix

### CODE REVIEW

**Comment by natemoo-re** [comment]
!bench

**Comment by matthewp** [comment]
I really love that command. Just need to clean up the output a little bit.

File: packages/astro/src/runtime/server/render/page.ts [hunk 1]
```diff
@@ -19,6 +20,29 @@ function nonAstroPageNeedsHeadInjection(pageComponent: NonAstroPageComponent): b
...
+async function iterableToHTMLBytes(
+	result: SSRResult,
+	iterable: ComponentIterable,
+	onDocTypeInjection?: (parts: HTMLParts) => Promise<void>
+): Promise<Uint8Array> {
+	const parts = new HTMLParts();
+	let i = 0;
+	for await (const chunk of iterable) {
+		if (isHTMLString(chunk)) {
+			if (i === 0) {
```
Comment (aFuzzyBear) [inline packages/astro/src/runtime/server/render/page.ts:32]: I cant see where `i` is being incremented within the for loop, it reads *forgive me, as a unnecessary conditional check, I mean its perhaps somewhere but I just cant see it from here,

**Review by aFuzzyBear — COMMENTED** [review]
Hey Matt, I am sorry I know this is merged, but I was taking a wee look over this, just doing some studying, and I got stumped on a few things. predomately with the `iterableToHTMLBytes` now Im not pretending to understand whats going on, kinda got a wee sense, but two things struck out to me. 
1) The logic is very nested, having looked at it I was wondering if it could be less deep
```diff
for await (const chunk of iterable) {
-		if (isHTMLString(chunk)) {
-			if (i === 0) {

-				if (!/<!doctype html/i.test(String(chunk))) {
+		if ( isHTMLString(chunk)
+             && i === 0
+             && !/<!doctype html/i.test(String(chunk))
                ) { parts.append('<!DOCTYPE html>\n', result);
					if(onDocTypeInjection) {
						await onDocTypeInjection(parts);
					}
				}
			}
		}
```
Also what is `i` doing exactly, I couldnt see where it gets incremented,

Sorry to take your time with this,

**Comment by matthewp** [comment]
Yeah we can talk about it. Are you investigating because you've encountered a bug you think was caused by this?

**Comment by aFuzzyBear** [comment]
no no, I was doing a wee bit of studying last night, if you got the time dude Id love to understand this whole thing alot better

**Comment by matthewp** [comment]
There was a bug that came about from this, fixed here: https://github.com/withastro/astro/pull/5035

**Pull request merged**